src/components/Footer/index.js

```diff
@@ -1,15 +1,270 @@
+    <>
+      <Typography
+        color="text.primary"
+        sx={{ mt: 4 }}
+        variant="subtitle1"
+        component="div"
+      >
+        {title}
+      </Typography>
+      <List>
+        {menu[menuItem].map((item) => {
+          if (item.type === "internal") {
+            return (
+              <Link
+                to={
+                  item.id
+                    ? interpolatePath(PATHS.PATHWAY_COURSE, {
+                        pathwayId: item.id,
+                      })
+                    : item.link
+                }
+                style={{
+                  textDecoration: "none",
+                }}
+              >
+                <Typography
+                  variant="body2"
+                  color="text.primary"
+                  sx={{ pb: "5px" }}
+                >
+                  {item.title}
+                </Typography>
+              </Link>
+            );
+          } else {
+            return (
+              <a
+                style={{
+                  textDecoration: "none",
+                }}
+                target="_blank"
+                href={item.link}
+              >
+                <Typography
+                  variant="body2"
+                  color="text.primary"
+                  sx={{ pb: "5px" }}
+                >
+                  {item.title}
+                </Typography>
+              </a>
+            );
+          }
+        })}
+      </List>
+    </>
+  );
+};
+
+function FooterIcon(props) {
+  const classes = useStyles();
+  return (
+    <Box className={classes.image}>
+      <img
+        src={require("./asset/" + props.name + ".svg")}
+        alt={props.alt || "image of " + props.name}
+        loading="lazy"
+      />
+    </Box>
+  );
+}
 
 function Footer() {
+  const classes = useStyles();
+  const [showHeader, setShowHeader] = React.useState(true);
+  const dispatch = useDispatch();
+  const { data } = useSelector((state) => state.Pathways);
+
+  useEffect(() => {
+    dispatch(pathwayActions.getPathways());
+  }, [dispatch]);
+
+  data &&
+    data.pathways.forEach((pathway) => {
+      menu.LearningTracks.forEach((item) => {
+        if (pathway.code === item.code) {
+          item["id"] = pathway.id;
+        }
+      });
+    });
+
+  useEffect(() => {
+    if (
+      window.location.pathname.split("/").includes("course-content") ||
+      window.location.pathname.split("/").includes("login") ||
+      window.location.pathname.split("/").includes("profile")
+    ) {
+      console.log("here");
+      setShowHeader(false);
+    }
+  }, []);
+
+  const history = useHistory();
+  history.listen((location, action) => {
+    if (
+      location.pathname.split("/").includes("course-content") ||
+      location.pathname.split("/").includes("login") ||
+      location.pathname.split("/").includes("profile")
+    ) {
+      console.log("not in header");
+      setShowHeader(false);
+    } else {
+      setShowHeader(true);
+    }
+  });
```

Comment:
I'd avoid hardcoding references to parts of the paths here because the paths might change.  For example, if `PATHWAY_COURSE_CONTENT` changed from a starting prefix of `/course-content/` to `course_content`, the footer would start to display on these pages.

Additionally, is it possible that in the future, you'd add a path where the pathname could include one of these strings and yet you'd still want to show the footer?  For example, is it possible you might introduce some path `/companyName/profile/` where you'd want to show the profile of some company but still show the footer?

Finally, it doesn't seem to be a good React practice to have a child component (i.e., the Footer) decide whether it should be appearing in a parent component.  So, this is how I might handle the situation.
1. Introduce a constant Array of paths for hiding the footer:
`const HIDE_FOOTER_PATHS = ['PATHWAY_COURSE_CONTENT', 'LOGIN', 'PROFILE'];`
2. Introduce a more robust function for matching specific paths:
```
export const matchPath = (pathConstant, path) =>
    new RegExp(pathConstant.replace(/:(\w*)/g, "(\\w*)")).test(path);
```
3. In the App component, include (and the suggested change):
```
  const history = useHistory();
  const [showFooter, setShowFooter] = React.useState(true);
  showFooter = true;
  history.listen((location) => {
    setShowFooter(!HIDE_FOOTER_PATHS.some(pathToHide => matchPath(pathToHide, location.pathname));
  });
```

Reply:
It'd actually probably be better to decide whether to display `<Footer>` in the Routing component since this is where the logic of which components to display for each path is housed.  Then you should also be able to avoid using the quick `matchPath` function I defined.  But I think that the solution I proposed above should be ok for now.

Reply:
We didn't get this solution. Where to do this?